LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+}
+```
+
+The `Tuna` asset is uniquely identified by an id.
+It also has a weight, which is limited between 500 grams and 1 million grams (a tonne).
+The largest tuna rarely exceed 800 kg.
+
+To specify the `Status` of the Tuna, that can be either `CAUGHT` or `PURCHASED`, we can define an enumerated type `Enum`, which specifies a type that can assume a limited number of values.
+
+```
+enum FishStatus {
+    o CAUGHT
+    o PURCHASED
+}
+```
+
+Then, we define the `Transaction`, to change the ownership of the Tuna from a `Fisher` to a `RestaurantOwner`.
+```
+transaction SellTuna {
+    --> Tuna tuna
+    --> RestaurantOwner restaurantOwner
+}
+```
+
+Finally, we define the `Event` to generate after the SellTuna transaction is executed:
+```
+event TunaSale {
+    o String tunaId
+    o String restaurantName
+}
+```
+
+## 4) Developing Transaction Logic
+The transaction logic is specified in the file `lib/logic.js`.
+
+We start by defining the same namespace specified in the modeling language file.
+
+```
+'use strict';
+/**
+ * Defining the namespace for the business network
+ */
+const NS = 'org.tuna';
+```
+
+The transaction logic is defined in a function that accepts the Transaction `SellTuna` as input parameter.
+```
+/**
+* Transfer tuna from one owner to another
+* @param {org.tuna.SellTuna} tx - The transferTuna transaction
+* @transaction
+*/
+async function sellTuna(tx) {
+```
+
+Next, the registries related to the Asset `Tuna` and the Participant `Restaurant Owner` are instantiated:
+```
+    // Get asset registry for Tuna
+    const tunaRegistry = await getAssetRegistry(NS + '.Tuna');
+
+    // Get participant registry for Individuals
+    const restaurantOwnerRegistry = await getParticipantRegistry(NS + '.RestaurantOwner');
+```
+
+Next, we have to verify that the `status` of the Tuna is `CAUGHT`.
+This is to make sure that a `Tuna` already sold cannot be sold again.
+```
+    // Make sure the tuna status is CAUGHT and not PURCHASED
+    if (tx.tuna.status !== 'CAUGHT') {
+        throw new Error(`Tuna with id ${tx.tuna.getIdentifier()} is not in CAUGHT status`);
+    }
+```
+
+Retrieve the id of the `RestaurantOwner` from the Transaction.
+```
+    // Get restaurantOwner ID
+    const restaurantOwnerId = tx.restaurantOwner.getIdentifier();
+```
+
+Next, we verify that the `RestaurantOwner` exists
+```
+    // Make sure that restaurantOwner exists
+    const restaurantOwner = await restaurantOwnerRegistry.get(restaurantOwnerId);
+    if (!restaurantOwner) {
+        throw new Error(`RestaurantOwner with id ${restaurantOwnerId} does not exist`);
+    }
+```
+
+Now we can update the `owner` of the Tuna:
+```
+    // Update tuna with new owner
+    tx.tuna.owner = tx.restaurantOwner;
+```
+
+Update the `status` of the Tuna:
+```
+    tx.tuna.status = 'PURCHASED';
+```
+
+Update the record of the Tuna in the asset registry:
+```
+    // Update the asset in the asset registry.
+    await tunaRegistry.update(tx.tuna);
+}
```

Comment:
This } is in the wrong spot with the event creation below.

Reply:
Fixed